x-pack/plugins/infra/public/components/logging/log_analysis_setup/process_step/process_step.tsx

```diff
@@ -30,6 +30,7 @@ interface ProcessStepProps {
   setUp: () => void;
   setupStatus: SetupStatus;
   viewResults: () => void;
+  onClose: () => void;
```

Comment:
Why should this step be aware that it's inside of something that can be "closed"? Shouldn't `viewResults()` as passed by the parent already do everything that is required to go to the full results view?

Reply:
`viewResults` comes from the `useLogEntry*Setup` hook, which knows nothing about the flyout.

I think the "View results" button should close the flyout. I can either pass the `onClose` callback to the step factory function, or wrap it up 1-2 components higher.

Reply:
But there's a parent component in between that could perform the composition of

```
  const viewResultsAndClose = useCallback(() => {
    viewResults();
    onClose();
  }, [viewResults, onClose]);
```

and just pass the result in here. That way the knowledge about being in a flyout wouldn't leak in here.